ClangFormat/TRVSCodeFragment.m

```diff
@@ -34,14 +34,57 @@ + (instancetype)fragmentUsingBlock:
 - (instancetype)initWithBuilder:(TRVSCodeFragmentBuilder *)builder {
   if (self = [super init]) {
     _string = [builder.string copy];
-    _range = builder.range;
+    _textRangePreFormat = builder.textRange;
     _fileURL = builder.fileURL;
   }
   return self;
 }
 
+// Find the smallest possible part that is different from before formatting,
+// so we don't have to replace the entire file.
+- (void)updateRangeToReplace:(NSString *)formattedDoc {
+  NSUInteger originalLen = _string.length;
+  NSUInteger formattedLen = formattedDoc.length;
+  NSRange originalSelection = _textRangePreFormat;
+  NSRange rangeInOriginalDoc = NSMakeRange(0, _string.length);
+  NSRange rangeInFormattedDoc = NSMakeRange(0, formattedDoc.length);
+
+  // Find the left boundary.
+  NSUInteger i;
+  for (i = 0; i < _textRangePreFormat.location - 1; ++i) {
```

Comment:
It looks like the `_textRangePreFormat.location - 1` is the culprit behind the issues I posted to this PR. In my (limited) local tests, removing the `-1` makes this code behave as expected.